src/init/starship.fish

```diff
@@ -1,19 +1,32 @@
+function __starship_set_job_count --description 'Set STARSHIP_JOBS using fish job groups (or legacy PIDs if toggled)'
+    # To force legacy behavior (process PIDs), set this variable to "false":
+    #   set -g __starship_fish_use_job_groups "false"
+    if test "$__starship_fish_use_job_groups" = "false"
+        # Legacy behavior: counts PIDs (may overcount pipelines with terminated producers)
+        set -l __count (jobs -p 2>/dev/null | count)
+    else
+        # Default behavior: count job groups
+        set -l __count (jobs -g 2>/dev/null | count)
+    end
+    set -gx STARSHIP_JOBS $__count
```

Comment:
```suggestion
    set -g STARSHIP_JOBS $__count
```

Also, in case that doesn't cause issues with counting, I would prefer doing the `set` in the `prompt*` function (e.g. `set STARSHIP_JOBS (count (__starship_list_jobs))`), but you can keep it like this to avoid undue changes.